### ISSUE TITLE
InstalledPackage should take over the existing App CR

### ISSUE THREAD

**Issue by blc1996** [carvel-dev/kapp-controller#184]
**Describe the problem/challenge you have**

Currently, we use App CR to deploy core addons like CNI and CPI. We want to migrate these core addons to packaging API. 

During the cluster upgrade, core addons like CNI and CPI should continue to work without interrupt. That is when we apply the InstalledPackge in the upgraded cluster, the old App CR should be seamlessly taken over by the newly applied InstalledPackage.

**Describe the solution you'd like**
InstalledPackage should take over the existing App CR if they have the same namespaced name.

**Anything else you would like to add:**
[Additional information that will assist in solving the issue.]

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by danielhelfand** [comment]
Thanks for opening this @blc1996. One thing we should consider with this is the case when a user doesn't expect this to happen (i.e. a user has an App CR created but doesn't want an InstalledPackage with the same name of the App to take over the already created App). 

We should maybe consider adding a property to the App CR spec so users can avoid an InstalledPackage using the already created App CR. This would cause an error to be returned saying the InstalledPackage cannot have the same name of an already existing App. Question would be what the defaults would be for that new App spec property (i.e. should kapp-controller allow the take over by an InstalledPackage by default or should it error out initially by default).

**Comment by blc1996** [comment]
Hi @danielhelfand , thanks for replying. I tried to use the InstalledPackage to take over the App CR of the same namespaced name. It seems that the current default behavior is that the App CR will be taken over by the InstalledPackage, and this is what we want. We synced with @cppforlife , and according to him, this should be expected and some test could be added in Carvel side to ensure this behavior.

If you guys decide to add that property, it would be more convenient for us to default to allow taking over.

Thanks!
